<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>810</width>
    <height>1035</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QRadioButton" name="RadioButton_MainAxisX">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>280</y>
      <width>92</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>X</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="RadioButton_MainAxisY">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>280</y>
      <width>92</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="RadioButton_MainAxisZ">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>280</y>
      <width>92</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>280</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Main Axis:</string>
    </property>
   </widget>
   <widget class="QFrame" name="FrameCurve">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>150</y>
      <width>821</width>
      <height>291</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QSlider" name="Slider_JointCount">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>90</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="minimum">
      <number>2</number>
     </property>
     <property name="maximum">
      <number>100</number>
     </property>
     <property name="value">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="NumberLabel_JointCount">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>90</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>2</string>
     </property>
    </widget>
    <widget class="QFrame" name="Frame_JointCount">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>90</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Raised</enum>
     </property>
    </widget>
    <widget class="QLabel" name="Label_JointCount">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>90</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint Count:</string>
     </property>
    </widget>
    <widget class="QLabel" name="Title_Curve">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Sans Serif Collection</family>
       <pointsize>12</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Ribbon Joint Guide Curve</string>
     </property>
    </widget>
    <widget class="QPushButton" name="Button_Curve">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>230</y>
       <width>211</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate NURBS Curve</string>
     </property>
    </widget>
    <widget class="QLabel" name="Label_JointPrefix">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>52</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Joint Prefix:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="LineEdit_JointPrefix">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>45</y>
       <width>113</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>JNT</string>
     </property>
    </widget>
    <widget class="Line" name="line_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>280</y>
       <width>811</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>170</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Curve Length:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="Spinbox_CurveLength">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>165</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>1000</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <zorder>Frame_JointCount</zorder>
    <zorder>Slider_JointCount</zorder>
    <zorder>NumberLabel_JointCount</zorder>
    <zorder>Label_JointCount</zorder>
    <zorder>Title_Curve</zorder>
    <zorder>Button_Curve</zorder>
    <zorder>Label_JointPrefix</zorder>
    <zorder>LineEdit_JointPrefix</zorder>
    <zorder>line_5</zorder>
    <zorder>label_2</zorder>
    <zorder>Spinbox_CurveLength</zorder>
   </widget>
   <widget class="QFrame" name="Frame_Ribbon">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>440</y>
      <width>811</width>
      <height>591</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QPushButton" name="Button_RibbonRig">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>500</y>
       <width>211</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Generate Ribbon Rig</string>
     </property>
    </widget>
    <widget class="QLabel" name="Title_Ribbon">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Sans Serif Collection</family>
       <pointsize>12</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Ribbon Rig Generation</string>
     </property>
    </widget>
    <widget class="QLabel" name="SubTitle_FKDeform">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>FK Deform:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_Roll">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>180</y>
       <width>78</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Roll</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="RadioButton_RollX">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>216</y>
       <width>92</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>X</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>216</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Roll Axis:</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="RadioButton_RollY">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>240</x>
       <y>216</y>
       <width>92</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_Ripple">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>250</y>
       <width>78</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Ripple</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>430</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Ribbon Deformer:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_Twist">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>460</y>
       <width>78</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Twist</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_Sine">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>460</y>
       <width>78</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Sine</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_Wire">
     <property name="geometry">
      <rect>
       <x>250</x>
       <y>460</y>
       <width>78</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Bend</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>48</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>FK:</string>
     </property>
    </widget>
    <widget class="Line" name="line">
     <property name="geometry">
      <rect>
       <x>-10</x>
       <y>290</y>
       <width>801</width>
       <height>20</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLineEdit" name="LineEdit">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>66</y>
       <width>113</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>FK</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>42</x>
       <y>76</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>FK CTRL Prefix:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="CheckBox_FKDriverCTRL">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>110</y>
       <width>251</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string> Generate FK Driver CTRL</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>320</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Ribbon:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>42</x>
       <y>350</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Ribbon Prefix:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="LineEdit_RibbonPrefix">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>343</y>
       <width>113</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Ribbon</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_21">
     <property name="geometry">
      <rect>
       <x>42</x>
       <y>385</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Ribbon Width:</string>
     </property>
    </widget>
    <widget class="QFrame" name="Frame_RibbonWidth">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>390</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Raised</enum>
     </property>
     <widget class="QLabel" name="NumberLabel_RibbonWidth">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>49</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </widget>
    <widget class="QSlider" name="Slider_RibbonWidth">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>390</y>
       <width>160</width>
       <height>16</height>
      </rect>
     </property>
     <property name="maximum">
      <number>40</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QRadioButton" name="RadioButton_RollZ">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>216</y>
       <width>92</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Z</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>60</y>
      <width>811</width>
      <height>91</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Sans Serif Collection</family>
       <pointsize>12</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Target Model</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="LineEdit_ModelName">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>40</y>
       <width>113</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>tentacle</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>131</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Model Name:</string>
     </property>
    </widget>
    <widget class="Line" name="line_2">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>80</y>
       <width>811</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QFrame" name="frame_6">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-40</y>
      <width>811</width>
      <height>101</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Shadow::Raised</enum>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>40</y>
       <width>351</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Sans Serif Collection</family>
       <pointsize>14</pointsize>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Ultimate Ribbon Rig</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>600</x>
       <y>60</y>
       <width>49</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>v 1.0.0</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>80</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>false</bold>
      </font>
     </property>
     <property name="text">
      <string>Created by Jim Zhao</string>
     </property>
    </widget>
    <widget class="Line" name="line_4">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>100</y>
       <width>811</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <zorder>Frame_Ribbon</zorder>
   <zorder>FrameCurve</zorder>
   <zorder>RadioButton_MainAxisX</zorder>
   <zorder>RadioButton_MainAxisY</zorder>
   <zorder>RadioButton_MainAxisZ</zorder>
   <zorder>label</zorder>
   <zorder>frame_4</zorder>
   <zorder>frame_6</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>810</width>
     <height>17</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>Slider_JointCount</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>NumberLabel_JointCount</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>229</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>364</x>
     <y>120</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CheckBox_Roll</sender>
   <signal>toggled(bool)</signal>
   <receiver>RadioButton_RollZ</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>702</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CheckBox_Roll</sender>
   <signal>toggled(bool)</signal>
   <receiver>RadioButton_RollX</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>532</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>CheckBox_Roll</sender>
   <signal>toggled(bool)</signal>
   <receiver>RadioButton_RollY</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>504</y>
    </hint>
    <hint type="destinationlabel">
     <x>285</x>
     <y>532</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>Slider_RibbonWidth</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>NumberLabel_RibbonWidth</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>730</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>730</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
